### PR TITLE
remote stats fixes

### PR DESCRIPTION
**Pull request by vrothberg**
* Write OK header once
* Detect closed client connections

@baude @jwhonce PTAL

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
but definitely want a @baude and/or @jwhonce head nod too.

File: pkg/api/handlers/compat/containers_stats.go [hunk 1]
```diff
@@ -175,11 +191,18 @@ func StatsContainer(w http.ResponseWriter, r *http.Request) {
 			Networks: net,
 		}
 
-		utils.WriteJSON(w, http.StatusOK, s)
+		if err := coder.Encode(s); err != nil {
+			utils.InternalServerError(w, err)
```
Comment (jwhonce) [inline pkg/api/handlers/compat/containers_stats.go:195]:
```suggestion
			logrus.Warnf(...)
```
You already sent 200 to client  you cannot send a 500 here.

**Review by jwhonce — CHANGES_REQUESTED** [review]

File: pkg/api/handlers/compat/containers_stats.go (hunk 1 above)
Comment (vrothberg) [reply pkg/api/handlers/compat/containers_stats.go]: Thanks for catching! I changed all other occurrences as well (but used Errorf).

**Pull request merged**